cpu/nrf5x_common/radio/nrfmin/nrfmin.c

```diff
@@ -401,6 +402,11 @@ static int nrfmin_init(netdev_t *dev)
         my_addr ^= cpuid[i] << (8 * (i & 0x01));
     }
 
+    /* the radio need the external HF clock source to be enabled */
+    /* @todo    add proper handling to release the clock whenever the radio is
+     *          idle */
+    clock_hfxo_request();
```

Comment:
You can turn it off in `goto_target_state()` when `target_state == STATE_OFF`

Reply:
I actually left this out intentionally for a follow up PR, as this would extend the testing scope quite significantly for this PR. Same for the 802154 driver, but there the vantage point seemed a little more obvious to me. But if you volunteer to test this, I can do it :-)

Reply:
Looking at the code with today's knowledge, it also seems to me that the `STATE_OFF` state is broken in general: when the radio is put into OFF mode (`NRF_RADIO->POWER = 0`), it looses all its register state and needs to be re-initialized. So the simple call to `NRF_RADIO->POWER = 1` in `nrfmin_set_state()` might be missleading in that regard. However, I currently don't have the time to do further investigations on this topic. So I think it is best to do any tinkering with the clocks apart from the inserted request, and verify the complete low-power mechanisms of the driver in a separate issue/PR as initially proposed.